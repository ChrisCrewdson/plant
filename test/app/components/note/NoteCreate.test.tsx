export {}; // To get around: Cannot redeclare block-scoped variable .ts(2451)

const React = require('react');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;
const {
  MemoryRouter,
} = require('react-router-dom');
const { Provider } = require('react-redux');
const NoteCreate = require('../../../../app/components/note/NoteCreate');
const store = require('../../../../app/store');
const App = require('../../../../app/components/App');

const muiTheme = getMuiTheme(lightBaseTheme);

describe('NoteCreate', () => {
  const storeDispatch = store.dispatch;

  beforeAll(() => {
    store.dispatch = () => {};
  });

  afterAll(() => {
    store.dispatch = storeDispatch;
  });

  test('should render a NoteCreate', () => {
    const interimNote = {
      note: 'test note text',
    };

    const plant = {
      _id: 'p-1',
    };

    const plants = {
      'p-1': {
        _id: 'p-1',
      },
    };

    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <Provider store={store}>
          <App>
            <MemoryRouter>
              <NoteCreate
                dispatch={storeDispatch}
                userCanEdit
                interimNote={interimNote}
                plant={plant}
                plants={plants}
                locationId="l-1"
              />
            </MemoryRouter>
          </App>
        </Provider>
      </MuiThemeProvider>);
    const badgeTotals = component.toJSON();
    expect(badgeTotals).toMatchSnapshot();
  });
});
