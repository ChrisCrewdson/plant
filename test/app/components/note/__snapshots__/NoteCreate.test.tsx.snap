// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCreate should render a NoteCreate 1`] = `
<div
  className="react-root"
>
  <div>
    <div
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      <button
        className="MuiButtonBase-root MuiFab-root MuiFab-secondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Create Note"
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            style={
              Object {
                "fontSize": "3em",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
